### ISSUE TITLE
Linestring decoding fails in non-vectorized path

### ISSUE THREAD

**Issue by springmeyer** [maplibre/maplibre-tile-spec#136]
Something is broken in linestring decoding in the non-vectorized path. This is not fixed in #129.

```
› java -jar ./build/libs/encode.jar -mvt ../test/fixtures/omt/10_532_684.mvt -decode -advanced  
Time elapsed for encoding: 473 milliseconds
java.lang.IllegalArgumentException: Invalid number of points in LineString (found 1 - must be 0 or >= 2)
	at org.locationtech.jts.geom.LineString.init(LineString.java:89)
	at org.locationtech.jts.geom.LineString.<init>(LineString.java:79)
	at org.locationtech.jts.geom.GeometryFactory.createLineString(GeometryFactory.java:621)
	at org.locationtech.jts.geom.GeometryFactory.createLineString(GeometryFactory.java:612)
	at com.mlt.decoder.GeometryDecoder.decodeGeometry(GeometryDecoder.java:146)
	at com.mlt.decoder.MltDecoder.decodeMlTile(MltDecoder.java:77)
	at com.mlt.tools.Encode.main(Encode.java:283)
```

But this works:

```
› java -jar ./build/libs/encode.jar -mvt ../test/fixtures/omt/10_532_684.mvt -decode -advanced -vectorized
Time elapsed for encoding: 459 milliseconds
Time elapsed for decoding: 15 milliseconds
```

**Comment by springmeyer** [comment]
After reviewing the current status of decoding across a large set of diverse tiles: this error is by far the most common. So I think this is the highest priority to fix. @mactrem is you have time tomorrow or Sat, would love your eyes on this one.

**Comment by springmeyer** [comment]
Now that I've got the JS decoder working on geometries (#157), I'm also now hitting this, or what I think is this bug.

In javascript the error is:
```
/Users/dane/projects/maplibre-tile-spec/js/src/data/Geometry.ts:26
            this.coordinates.push([vertex.x, vertex.y]);
                                          ^
TypeError: Cannot read properties of undefined (reading 'x')
    at new LineString (/Users/dane/projects/maplibre-tile-spec/js/src/data/Geometry.ts:26:43)
    at GeometryFactory.createLineString (/Users/dane/projects/maplibre-tile-spec/js/src/data/Geometry.ts:137:16)
    at Function.decodeGeometry (/Users/dane/projects/maplibre-tile-spec/js/src/decoder/GeometryDecoder.ts:127:69)
    at Function.decodeMlTile (/Users/dane/projects/maplibre-tile-spec/js/src/decoder/MltDecoder.ts:57:50)
```

What is happening is that `numVertices` is `undefined` and we end up creating an Array like `new Array(undefined)` and that leads to `[undefined]`.

In java `getLineString(vertexBuffer, vertexBufferOffset, numVertices, false);` is returning:

> [(3357.0, 306.0, NaN)]

Could this be an encoding bug?

**Comment by springmeyer** [comment]
Just hit another error in the java decoder that looks like it might be a version of this bug:

For this tile: `./test/expected/omt/13_4265_5467.mlt`:

```
java.lang.ArrayIndexOutOfBoundsException: Index -5 out of bounds for length 4
	at com.mlt.metadata.stream.StreamMetadata.decode(StreamMetadata.java:66)
	at com.mlt.metadata.stream.StreamMetadataDecoder.decode(StreamMetadataDecoder.java:8)
	at com.mlt.decoder.GeometryDecoder.decodeGeometryColumn(GeometryDecoder.java:28)
	at com.mlt.decoder.MltDecoder.decodeMlTile(MltDecoder.java:77)
	at com.mlt.tools.Encode.main(Encode.java:269)
```

**Comment by springmeyer** [comment]
Solved by #158
